m365-developer-proxy-plugins/m365-developer-proxy-plugins.csproj

```diff
@@ -7,6 +7,7 @@
     <EnableDynamicLoading>true</EnableDynamicLoading>
     <RunPostBuildEvent>OnOutputUpdated</RunPostBuildEvent>
     <Version>0.12.0</Version>
+    <NoWarn>CS1998</NoWarn>
```

Comment:
Personally I'd rather have explicit pragma comments disabling specific warnings on this one rather than potentially hiding a bunch cases where there should be `await` or the method should not use async.

Reply:
I added this one, because the plugins project if full with this one. I thought one row in the project vs several line ignored on all the occasions.

How about adding the Release condition? Then you still have the warnings during development and they won't show up in the build pipeline.

Reply:
That's better. I've just seen these kind of blanket suppressions for a project cause more problems than having the warnings visible to be comfortable with suppressing them entirely

Reply:
```suggestion
    <PropertyGroup Condition="'$(Configuration)'=='Release'">
      <NoWarn>CS1998</NoWarn>
    </PropertyGroup>
```

Reply:
Actually, it needs a separate PropertyGroup with the condition, nesting like the suggestion doesn't work